### DISCUSSION TITLE
SP4000 - not able to connect and after some advice

### DISCUSSION THREAD

**Discussion by Agent6-6-6** [ParadoxAlarmInterface/pai#448] [General]
Hi All

I have a paradox SP4000 alarm with TM50 touchpad and hoping to integrate with home assistant. We had it installed about 10 years, and have not had to touch it since the install. Original installer is still in business but have not approached them yet as not really sure what to even ask them or if it is some setting somewhere.
 
On PCB there is a sticker with '5.16' which I assume is maybe the firmware version it came with (and still has). Noted under the compatibility it noted 5.20 had been tried successfully, but not sure if that is a minimum version required. Unclear without being able to connect how the firmware would even be updated.

I figured out the installer code by starting at 0000 and incrementing on the keypad until I figured installer code out as being 1410. The maintenance code I set to 0000 after hacking the installer code (was blank previously). 

Trying to connect via ESP32 serial via IP but get the following, unable to connect using babyware so really unclear why it may not be working:-
![Screenshot 2024-03-30 172015](https://github.com/ParadoxAlarmInterface/pai/assets/36808164/caf80ebb-5e0a-473a-8cbd-d1d206291a17)

Hardware is ESP32 connected through buck converter to get the 5V from aux connection via serial connector. Initially didn't have the TX and RX through a level shifter to get the data to 3.3V input (unsure if this has damaged the ESP32 as not 5V tolerant on the data lines I've since found out). Can telnet to the device on wifi, suggesting ESP32 is opening the connection, but unable to connect beyond this. Tried swapping TX/RX to ESP32 but same behaviour. 

The issue is cannot seem to connect at all, and not really sure if it is something hardware or software related. Only tried a single type of ESP32 board using the code at https://github.com/espressif/arduino-esp32/blob/master/libraries/WiFi/examples/WiFiTelnetToSerial/WiFiTelnetToSerial.ino modified to correct wifi credentials and baud rate to 9600 (vs 115200) as understand SP4000 is 9600.

Current yaml in Home Assistant is as follows if there's something there that's clearly wrong (but given I cannot connect via babyware either seems like something else):-
```LOGGING_LEVEL_CONSOLE: 20
LOGGING_LEVEL_FILE: 40
CONNECTION_TYPE: IP
SERIAL_PORT: /dev/ttyUSB0
SERIAL_BAUD: 9600
IP_CONNECTION_HOST: 192.168.69.124
IP_CONNECTION_PORT: 23
IP_CONNECTION_PASSWORD: paradox
KEEP_ALIVE_INTERVAL: 10
LIMITS:
  zone: auto
  user: 1-10
  door: ""
  pgm: 1-5
  partition: auto
  module: ""
  repeater: ""
  keypad: ""
  key-switch: ""
SYNC_TIME: true
SYNC_TIME_MIN_DRIFT: 120
PASSWORD: "1410"
MQTT_ENABLE: true
MQTT_HOST: core-mosquitto
MQTT_PORT: 1883
MQTT_KEEPALIVE: 60
MQTT_USERNAME: pai
MQTT_PASSWORD: xxxxx
MQTT_HOMEASSISTANT_AUTODISCOVERY_ENABLE: true
COMMAND_ALIAS:
  arm: partition all arm
  disarm: partition all disarm
MQTT_COMMAND_ALIAS:
  armed_home: arm_stay
  armed_night: arm_sleep
  armed_away: arm
  disarmed: disarm
HOMEASSISTANT_NOTIFICATIONS_EVENT_FILTERS:
  - live,alarm,-restore
  - trouble,-clock
  - live,tamper
HOMEASSISTANT_PUBLISH_PARTITION_PROPERTIES:
  - target_state
  - current_state
HOMEASSISTANT_PUBLISH_ZONE_PROPERTIES:
  - open
  - tamper
PUSHBULLET_CONTACTS: []
PUSHBULLET_EVENT_FILTERS:
  - live,alarm,-restore
  - trouble,-clock
  - live,tamper
PUSHOVER_EVENT_FILTERS:
  - live,alarm,-restore
  - trouble,-clock
  - live,tamper
PUSHOVER_BROADCAST_KEYS: []
SIGNAL_CONTACTS: []
SIGNAL_EVENT_FILTERS:
  - live,alarm,-restore
  - trouble,-clock
  - live,tamper
GSM_CONTACTS: []
GSM_EVENT_FILTERS:
  - live,alarm,-restore
  - trouble,-clock
  - live,tamper
IP_INTERFACE_ENABLE: true
IP_INTERFACE_PASSWORD: paradox
DUMMY_EVENT_FILTERS: []
IP_CONNECTION_BARE: true
MQTT_BASE_TOPIC: paradox
```

Read in a few replies serial can be locked, could this be the issue? If so can it be unlocked via reset to defaults or knowing the installer password? (was trying to avoid having to set everything up again, but if it's the only way then no worries)

Next step I was going to try was just uninstalling and starting the PAI setup again from scratch, but I feel something else is going on and that someone else probably has a suggestion on where I've gone wrong :)

Can provide anything further, more details on hardware etc if someone thinks that is the issue. May have something setup incorrectly, must admit the wiki instructions seem clear enough on the surface, but also somewhat vague in exact specifics, so I may have misinterpreted.

thanks!

**Comment by Agent6-6-6** [comment]
Anyone?

**Reply by MBRetrofit** [reply]
Did you solved? I am in the same position

**Reply by Agent6-6-6** [reply]
@MBRetrofit no didn't manage to fix it. SP4000 doesn't seem to have any menu to check if serial is locked (that I'm able to find) which seems to be the general consensus on this error as far as I was able to find out from a lot of reading. 

Currently looking at utilising Konnected on a spare esp32. But waiting on a relay. But managed to programme the SP4000 so I can remotely arm and disarm by shorting a zone input with a wire + resistor. Just need the relay to automate this via home assistant and esphome. May commit to ordering the konnected interface board once I know the solution is workable but it's many times the cost of a esp32 board, so I kind of wanted to make sure it really works.

**Comment by yozik04** [comment]
Sorry guys. I cannot help with ESP32 because I do not use one. I also do not have any MG/SP. You need to try to ask people who use ESP32. I think you will be able to find them in issues.

**Comment by MBRetrofit** [comment]
Although PAI seams to be a very strong addon, the use of ESP32/8266 should be the main feature, I think. 
The Paradox is obsolete and adding cheap HA functionality could revive this alarm.

Any panels, IP modules, relays is intended only for users that do not use HA.

And maybe because no entities will be broadcasted - the 8266 should be more than enough.
